language/sygus.tex

```diff
@@ -2155,27 +2158,47 @@ \subsection{Satisfying Semantic Specifications}
 The notion of satisfying semantic restrictions for background theories
 that are not standardized in the SMT-LIB standard are not covered by this document.
 
-A tuple of functions 
-$( \lambda X_1 \ldotp t_1, \ldots, \lambda X_n \ldotp t_n)$
-satisfies the semantic restrictions
-for functions-to-synthesize $( f_1, \ldots, f_n )$
-in conjecture $\exists f_1, \ldots, f_n \ldotp\Psi$ in background theory $T$
-if $\Psi$ is $T$-valid formula
-when $f_1, \ldots, f_n$ are defined to be terms
-whose semantics are given by the functions
-$\lambda X_1 \ldotp t_1, \ldots, \lambda X_n \ldotp t_n$.
+Before stating the formal definition for satisfying semantic specifications,
+we require the following definitions.
+Consider a synthesis conjecture $\exists f_1, \ldots, f_n \ldotp\Psi$
+in background theory $T$, and let
+$\alpha = ( \lambda X_1 \ldotp t_1, \ldots, \lambda X_n \ldotp t_n)$
+be an assignment for functions-to-synthesize $( f_1, \ldots, f_n )$
+whose grammars are $G_1, \ldots, G_n$.
+Let $\Psi_{\alpha}$ be the formula:
+\[
+\bigwedge_{i=1}^{n} \forall X_i f( X_i ) \teq t_i
```

Comment:
Another minor comment, but what are your thoughts on using:

```tex
\global\long\def\teq{\overset{T}{=}}
```

Just a bit more intuitive that this the infix equality predicate under theory `T`.

Reply:
I'd prefer not having this annotation, since its easy to understand equality as being polymorphic.

Reply:
I agree, it's not that major. The polymorphic part is fine, but I was suggesting the `T` overset just to distinguish this equality predicate from `=`.